chainservice/builder.go

```diff
@@ -33,23 +38,34 @@ import (
 	"github.com/iotexproject/iotex-core/p2p"
 	"github.com/iotexproject/iotex-core/pkg/log"
 	"github.com/iotexproject/iotex-core/state/factory"
-	"github.com/iotexproject/iotex-election/committee"
-	"github.com/pkg/errors"
-	"go.uber.org/zap"
-	"google.golang.org/protobuf/proto"
 )
 
-// Builder is a builder to build chainservice
-type Builder struct {
-	cfg config.Config
-	cs  *ChainService
+type (
+	// Builder is a builder to build chainservice
+	Builder struct {
+		cfg        config.Config
+		cs         *ChainService
+		stopHeight uint64 // for tools/timemachine
+	}
+
+	// BuilderOption sets Builder construction parameter
+	BuilderOption func(*Builder)
+)
+
+// WithStopHeightBuilderOption sets the stopHeight
+func WithStopHeightBuilderOption(stopHeight uint64) BuilderOption {
+	return func(svr *Builder) {
+		svr.stopHeight = stopHeight
+	}
 }
 
 // NewBuilder creates a new chainservice builder
-func NewBuilder(cfg config.Config) *Builder {
+func NewBuilder(cfg config.Config, opts ...BuilderOption) *Builder {
```

Comment:
`opts` is better to be used in function `Build`.

Reply:
ok. fixed